docs/ui/components/CommandMenu/CommandMenu.tsx

```diff
@@ -0,0 +1,105 @@
+import { SearchIcon, theme, XIcon } from '@expo/styleguide';
+import { Command } from 'cmdk';
+import { useEffect, useState } from 'react';
+import type { Dispatch, SetStateAction } from 'react';
+
+import { BarLoader } from './BarLoader';
+import { CommandFooter } from './CommandFooter';
+import { RNDirectoryItem, RNDocsItem, ExpoDocsItem, ExpoItem } from './Items';
+import { entries } from './expoEntries';
+import { searchIconStyle, closeIconStyle } from './styles';
+import type { ExpoItemType, RNDirectoryItemType, AlgoliaItemType } from './types';
+import { getExpoDocsResults, getRNDocsResults, getDirectoryResults, getItems } from './utils';
+
+import { CALLOUT } from '~/ui/components/Text';
+
+type Props = {
+  version: string;
+  open: boolean;
+  setOpen: Dispatch<SetStateAction<boolean>>;
+};
+
+export const CommandMenu = ({ version, open, setOpen }: Props) => {
+  const [loading, setLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [expoDocsItems, setExpoDocsItems] = useState<AlgoliaItemType[]>([]);
+  const [expoItems, setExpoItems] = useState<ExpoItemType[]>([]);
+  const [rnDocsItems, setRnDocsItems] = useState<AlgoliaItemType[]>([]);
+  const [directoryItems, setDirectoryItems] = useState<RNDirectoryItemType[]>([]);
+
+  const getExpoDocsItems = async () =>
+    getItems(query, getExpoDocsResults, setExpoDocsItems, version);
+  const getRNDocsItems = async () => getItems(query, getRNDocsResults, setRnDocsItems);
+  const getDirectoryItems = async () => getItems(query, getDirectoryResults, setDirectoryItems);
+
+  const getExpoItems = async () => {
+    setExpoItems(entries.filter(entry => entry.label.toLowerCase().includes(query.toLowerCase())));
+  };
+
+  const dismiss = () => setOpen(false);
+
+  const fetchData = () => {
+    Promise.all([getExpoDocsItems(), getRNDocsItems(), getDirectoryItems(), getExpoItems()]).then(
+      () => setLoading(false)
+    );
+  };
+
+  useEffect(() => {
```

Comment:
nitpick: in other codebases, we'd write this as:

```tsx
useEffect(function didUpdate() {
  // ...
}, [query]);
```

Or `queryDidUpdate`.

We name effects to make them more descriptive.

Reply:
In docs, we almost don't use this pattern. On around 50 use cases, we named only three, and all of them are related to `didMount` phase (with no dependencies on any props or state value), so I will leave this as-is.

Feel free to refactor all of those in the separate PR, if you would like to enforce this pattern in docs too.